tests/model_formsets/tests.py

```diff
@@ -1771,6 +1771,23 @@ def test_initial_form_count_empty_data(self):
         formset = AuthorFormSet({})
         self.assertEqual(formset.initial_form_count(), 0)
 
+    def test_prevent_editing_object_outside_of_queryset(self):
+        qs = Author.objects.bulk_create([
+            Author(name='Charles Baudelaire'),
+            Author(name='Walt Whitman')]
+        )
+        data = {
+            'form-TOTAL_FORMS': '1',
+            'form-INITIAL_FORMS': '1',
+            'form-0-id': str(qs[1].pk),
```

Comment:
The primary key attribute can only be retrieved on certain databases, this will not work on Oracle or MySQL.